src/main/java/org/eolang/jetbrains/EoCompileAction.java

```diff
@@ -0,0 +1,128 @@
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.project.Project;
+import java.util.Collections;
+import java.util.function.Supplier;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.idea.maven.execution.MavenRunConfigurationType;
+import org.jetbrains.idea.maven.execution.MavenRunnerParameters;
+import org.jetbrains.idea.maven.model.MavenExplicitProfiles;
+import org.jetbrains.idea.maven.model.MavenPlugin;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+import org.jetbrains.idea.maven.utils.actions.MavenActionUtil;
+
+/**
+ * Compile EO sources.
+ * @since 1.0
+ * @todo #20:30min Make compile action complete adding remaining compile actions
+ *  (assemble, transpile etc.)
+ * @todo #20:30min Create tests for the action
+ * @checkstyle LineLengthCheck (30 lines)
+ */
+public final class EoCompileAction extends AnAction {
+    @Override
+    public void actionPerformed(@NotNull final AnActionEvent action) {
+        final DataContext context = action.getDataContext();
+        try {
+            final Project project = EoCompileAction.supplyNoNull(
+                () -> MavenActionUtil.getProject(context),
+                () -> EoCompileAction.notifyCannotCompile("Select Idea project")
+            );
+            final MavenProject maven = EoCompileAction.supplyNoNull(
+                () -> MavenActionUtil.getMavenProject(context),
+                () -> EoCompileAction.notifyCannotCompile(
+                    String.format("Set up Maven project for %s", project.getName())
+                )
+            );
+            final MavenPlugin plugin = EoCompileAction.supplyNoNull(
+                () -> maven.findPlugin("org.eolang", "eo-maven-plugin"),
+                () -> EoCompileAction.notifyCannotCompile(
+                    "eo-maven-plugin is not configured in your pom.xml, see https://github.com/objectionary/eo/tree/master/eo-maven-plugin"
+                )
+            );
+            final MavenProjectsManager manager = EoCompileAction.supplyNoNull(
+                () -> MavenActionUtil.getProjectsManager(context),
+                () -> {
+                }
+            );
+            final MavenExplicitProfiles profiles = manager.getExplicitProfiles();
+            MavenRunConfigurationType.runConfiguration(
+                project,
+                new MavenRunnerParameters(
+                    true,
+                    maven.getDirectory(),
+                    maven.getFile().getName(),
+                    Collections.singletonList(
+                        String.format("org.eolang:eo-maven-plugin:%s:register", plugin.getVersion())
+                    ),
+                    profiles.getEnabledProfiles(),
+                    profiles.getDisabledProfiles()
+                ),
+                null
+            );
+        } catch (final IllegalStateException ignored) {
```

Comment:
@mximp oops.. why??

Reply:
@yegor256 as we don't want to propagate exceptions into the Idea UI. See my comment below.